data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
-        valid = repr_df.query("split == 'valid'").reset_index(drop=True)
-
-        split = {"train": train, "test": test, "valid": valid}
-
-    # create train and test dataframes
-    train_df = split["train"]
-    test_df = split["test"]
-    valid_df = split["valid"]
-    # add split columns to train and test dataframes
-    train_df["split"] = len(train_df) * ["train"]
-    test_df["split"] = len(test_df) * ["test"]
-    valid_df["split"] = len(valid_df) * ["valid"]
-
-    # merge train and test across all datasets
-    merge = pd.concat([train_df, test_df, valid_df], axis=0)
-    # rewrite data_clean.csv for each dataset
-    rewrite_data_with_splits(
-        csv_paths=paths_to_data,
-        repr_col=repr_col,
-        train_test_df=merge,
-        override=override,
-        check=check,
+
+def scaffold_split(
+    data_dir: Union[str, Path],
+    override: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+    seed: int = 42,
+    debug: bool = False,
+):
+    """Performs scaffold splitting on all datasets in the data_dir directory.
+    It serializes the validation and test smiles into `val_smiles.txt` and `test_smiles.txt`
+    in the current directory.
+
+    It is the first step of the splitting procedure as we will then perform a random split
+    on the remaining datasets on all other datasets. However, if they have smiles,
+    we will set all smiles that are in the validation and test sets to be in the validation and test sets.
+
+    Args:
+        data_dir (Union[str, Path]): The directory containing the directories with the datasets.
+        override (bool, optional): Whether to override the existing data_clean.csv files with the new ones.
+            Defaults to False. In this case, the old data_clean.csv files will be renamed to data_clean_old.csv.
+        train_frac (float, optional): The fraction of the data to be used for training. Defaults to 0.7.
+        val_frac (float, optional): The fraction of the data to be used for validation. Defaults to 0.15.
+        test_frac (float, optional): The fraction of the data to be used for testing. Defaults to 0.15.
+        seed (int, optional): The random seed. Defaults to 42.
+        debug (bool, optional): Whether to run in debug mode. Defaults to False.
+            In this case, only the first 5 datasets will be processed.
+    """
+    # make deterministic
+    np.random.seed(seed)
+    random.seed(seed)
+
+    all_yaml_files = get_yaml_files(data_dir)
+    if debug:
+        all_yaml_files = all_yaml_files[:5]
+    transformed_files = []
+    for file in all_yaml_files:
+        print(f"Processing {file}")
+        with open(file, "r") as f:
+            meta = yaml.safe_load(f)
+
+        if meta["name"] in to_scaffold_split:
+            # run transform.py script in this directory if `data_clean.csv` does not exist
+            # use this dir as cwd
+            run_transform(file)
```

Comment:
better check here if `meta.yaml` and `data_clean.csv` exist, see my comment above

Reply:
You mean instead of `run_transform`?